src/Microsoft.Diagnostics.Runtime/Utilities/DumpReader.cs

```diff
@@ -382,6 +384,186 @@ public static bool MiniDumpReadDumpStream(IntPtr pBase, MINIDUMP_STREAM_TYPE typ
                 return false;
             }
 
+            #region MiniDump Handles
+
+            public class MINIDUMP_HANDLES_STREAM
+            {
+                public MINIDUMP_HANDLES_STREAM(DumpPointer pStream, DumpReader contex)
+                {
+                    _pStream = pStream;
+                    _contex = contex;
+                }
+
+                DumpPointer _pStream;
+                DumpReader _contex;
+
+                /// <summary>
+                /// Read MiniDump handles from dump file
+                /// </summary>
+                /// <returns></returns>
+                public IEnumerable<DumpHandle> GetHandles()
+                {
+                    var handleData = _pStream.PtrToStructure<MINIDUMP_HANDLE_DATA_STREAM>();
+                    var dumpPointer = _pStream.Adjust(handleData.SizeOfHeader);
+
+                    if (handleData.SizeOfDescriptor == Marshal.SizeOf(typeof(MINIDUMP_HANDLE_DESCRIPTOR)))
+                    {
+                        for (int i = 0; i < handleData.NumberOfDescriptors; i++)
+                        {
+                            var descriptor = dumpPointer.PtrToStructure<MINIDUMP_HANDLE_DESCRIPTOR>();
+                            yield return GetHandleData(descriptor);
+                            dumpPointer = dumpPointer.Adjust((ulong)handleData.SizeOfDescriptor);
+                        }
+                    }
+                    else if (handleData.SizeOfDescriptor == Marshal.SizeOf(typeof(MINIDUMP_HANDLE_DESCRIPTOR_2)))
+                    {
+                        for (int i = 0; i < handleData.NumberOfDescriptors; i++)
+                        {
+                            var descriptor = dumpPointer.PtrToStructure<MINIDUMP_HANDLE_DESCRIPTOR_2>();
+                            yield return GetHandleData(descriptor, dumpPointer);
+                            dumpPointer = dumpPointer.Adjust((ulong)handleData.SizeOfDescriptor);
+                        }
+                    }
+                }
+
+                /// <summary>
+                /// Constructs handles wrapped class with MINIDUMP_HANDLE_DESCRIPTOR_2 struct data
+                /// </summary>
+                /// <param name="handle">minidump structure descriptor</param>
+                /// <param name="pointer">DumpPointer to stream</param>
+                /// <returns></returns>
+                private DumpHandle GetHandleData(MINIDUMP_HANDLE_DESCRIPTOR_2 handle, DumpPointer pointer)
+                {
+                    string objectName, typeName;
+                    typeName = objectName = null;
+
+                    if (handle.ObjectNameRva != 0)
+                    {
+                        objectName = _contex.GetString(new DumpNative.RVA() { Value = (uint)handle.ObjectNameRva });
```

Comment:
Why the cast to `uint`? Isn't it possible for the RVA to be greater than 4GB in large dumps?